pkg/rhos/gw-machineset.go

```diff
@@ -25,21 +25,21 @@ metadata:
     machine.openshift.io/cluster-api-cluster: {{.InfraID}}
     machine.openshift.io/cluster-api-machine-role: worker
     machine.openshift.io/cluster-api-machine-type: worker
-  name: {{.InfraID}}-submariner-gw-{{.Index}}
+  name: {{.InfraID}}-submariner-gw-{{.UUID}}
```

Comment:
Also, wont this change be applicable for the other cloud providers?

Reply:
The internal securtiy group is not added in the same way to the VM in all cloud providers , so this is Openstack specific. Moreover we are testing in OVNKubernetes with other cloud providers and it works.